From 43d2b491a53ccc347578390d5dbfedee0c580ea6 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Thu, 04 Mar 2021 11:58:57 +0200
Subject: [PATCH] [SW-20154] gaudi2_arc_gnu: change eROM recovery flow

If for some reason, the recovery binary from eROM fails
to execute due to CRC/SHA failure, dont return to main
flow. Instead stay in the execute pcie recovery function.

Change-Id: I835e3520a70da361e050f7789dbe764848dcc54b
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/common.c b/gaudi2_arc_gnu/common.c
index eb0c362..32fff06 100644
--- a/gaudi2_arc_gnu/common.c
+++ b/gaudi2_arc_gnu/common.c
@@ -312,6 +312,13 @@
 
 	/* Further, load the recovery bin from eROM */
 	ext_rom_load_erom_recovery();
+
+	/*
+	 * if we reach here, because of SHA/CRC error in execution of
+	 * eROM recovery, stay here, and don't return
+	 */
+	while (1)
+		;
 }
 
 void arcboot_rom_handle_tpm(void)
